### PR TITLE
#164046205 Setup React Router Version 4

### PR DESCRIPTION
**Pull request by RachelleMaina**
### What does this PR do?
It setups react router version 4.

### Description
Installs and configures the router by creating these folders: components, routes and css and these components: Header, Footer, Login and Home  in order to create a dummy route for login and the "/" page to test the configuration.

**Type of change**
- [x] chore(non-breaking change that sets up react router).

### Change log
- App.scss and index.scss files have been moved to the `css` folder
- App.js has been removed and replaced with `routes/index.js``

### How to manually test this
Clone the repo and checkout to this branch `ch-react-router-164046205`. Install the extra dependencies: `react-router-dom` and `react-bootstrap`  then start the app using `npm start`. On the browser you should see the navbar, a `Home` text and the footer. On the navbar, click `Login` to go to the Login component.

### PT
[#164046205](https://www.pivotaltracker.com/story/show/164046201)

### CODE REVIEW

**Review by abulojoshua1 — APPROVED** [review]
LGTM

**Review by abulojoshua1 — CHANGES_REQUESTED** [review]
I think `src/css/index.scss` should not be shown as a change in your PR, do you think it is possible to fix this.

**Comment by RachelleMaina** [comment]
> I think `src/css/index.scss` should not be shown as a change in your PR, do you think it is possible to fix this.

Yes. I have done this.

File: src/components/Header.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+const Header = () => {
+  return (
+    <Navbar expand="lg" className="navbar-custom">
+      <NavLink exact to="/" className="brand">
+        Author&apos;s Haven
+      </NavLink>
+      <Navbar.Toggle aria-controls="basic-navbar-nav" />
+      <Navbar.Collapse id="basic-navbar-nav">
+        <Nav className="ml-auto">
+          <NavLink to="/login" className="nav-link">
```
Comment (MandelaK) [inline src/components/Header.js]: In our mockups, we have another link for registration on the navbar, I believe it's titled "Get started" ... could you please include it in this component?

**Review by cob04 — CHANGES_REQUESTED** [review]
I am getting this error 
`Module not found: Can't resolve './css/index.scss' in '/Users/cobain/Andela/tester/ah-legion-frontend/src'`

Please move `index.scss` into the `css` directory to resolve this.

File: src/components/Header.js (hunk 1 above)
Comment (RachelleMaina) [reply src/components/Header.js]: As per the PT story, I only had to add the "/" and "login "routes. I think this will be done on the registration PR

**Comment by RachelleMaina** [comment]
> I am getting this error
> `Module not found: Can't resolve './css/index.scss' in '/Users/cobain/Andela/tester/ah-legion-frontend/src'`
> 
> Please move `index.scss` into the `css` directory to resolve this.

I do not think I am supposed to change that in this PR. I have however corrected the way it is imported. It should work now.

**Comment by MandelaK** [comment]
@RachelleMaina why did you delete `app.json`? I believe it's used to set up our pipelines.

**Comment by RachelleMaina** [comment]
> @RachelleMaina why did you delete `app.json`? I believe it's used to set up our pipelines.

I did not have an `app.json`. I did have an `App.js` though which was not necessary in its current location. I deleted that. I created` index.jsx` in the Routes/ directory which adds better functionality

**Comment by RachelleMaina** [comment]
> @RachelleMaina why did you delete `app.json`? I believe it's used to set up our pipelines.

I see it's present in the develop branch. let me see what I have done wrong.

**Comment by RachelleMaina** [comment]
> @RachelleMaina why did you delete `app.json`? I believe it's used to set up our pipelines.

I have corrected this.

**Review by cob04 — APPROVED** [review]
The routes work great.

**Review by ElMonstro — APPROVED** [review]
Approved

**Review by MandelaK — APPROVED** [review]

**Review by Etomovich — APPROVED** [review]

**Review by BrianSerem — APPROVED** [review]
What you have done here is good work, unless you add more changes, this is good to go.

**Review by abulojoshua1 — APPROVED** [review]
LGTM

**Pull request merged**